app/controllers/patients_controller.rb

```diff
@@ -38,6 +38,10 @@ def show
     @translations = Assessment.new.translations
 
     @history_types = History::HISTORY_TYPES
+
+    # Update viewer cache
+    @viewers = (Rails.cache.read("patient-view-#{@patient.id}")&.to_set || Set.new).delete(current_user.email)
+    Rails.cache.write("patient-view-#{@patient.id}", (@viewers + [current_user.email]).to_a, expires_in: 12.hours)
```

Comment:
Does the cache not allow you to leave it as a Set instead of switching between Array and Set types?

Reply:
Set isn't an enumerable type unfortunately. I do think there is a better way to go about this to ensure uniqueness (likely just using a hash). I'm going to revisit that later, but wanted something bare bones to go through UAT (to get look & feel feedback).

Reply:
If we don't really expect very many users to be viewing a single patient at the same time then we could maybe make this cleaner by keeping it as an array the entire time
```ruby
# Update viewer cache
key = "patient-view-#{@patient.id}"
@viewers = (Rails.cache.read(key) || []) - [current_user.email]
Rails.cache.write(key, @viewers + [current_user.email], expires_in: 12.hours)
```

Reply:
I'm not sure we can bet on a low number of users - I'd like to make this work for any amount - and with that in mind I'd like to make use of the quicker lookup/deletion Hash gives over searching an array.